vault/client.go

```diff
@@ -114,6 +125,21 @@ func (c *Client) GetSecret(secretEngine string, path string, keys []string, vers
 		return nil, fmt.Errorf("vaultNamespace field can not be used, because the VAULT_NAMESPACE environment variable is not set")
 	}
 
+	// Check whether the `path` matches with the allowed path. This allows the
+	// operator to add client-side restictions on top of the Vault (server-side)
+	// policies.
+	// For example:
+	//   c.allowedPath = '(common|{{namespace}})/.*'
+	// Restricts the operator to getting secrets from:
+	//   /secret/data/common
+	//   /secret/data/$secretNamespace/
+	if c.allowedPath != "" {
+		allowedPathRegex := regexp.MustCompile(strings.ReplaceAll(c.allowedPath, "{{namespace}}", secretNamespace))
```

Comment:
I'm not sure about the `{{namespace}}` or if sth. like `{% .Namespace %}` which is already used by the template feature would be better.

What do you think?

Reply:
Sure, open for any suggestion here.

Reply:
Then I would go with `{% .Namespace %}` for now.